src/wallet/signer.rs

```diff
@@ -504,10 +504,10 @@ fn sign_psbt_schnorr(
         Some(_) => keypair, // no tweak for script spend
     };
 
-    let sig = secp.sign_schnorr(
-        &Message::from_slice(&hash.into_inner()[..]).unwrap(),
-        &keypair,
-    );
+    let msg = &Message::from_slice(&hash.into_inner()[..]).unwrap();
+    let sig = secp.sign_schnorr(msg, &keypair);
+    secp.verify_schnorr(&sig, msg, &XOnlyPublicKey::from_keypair(&keypair))
```

Comment:
Confusingly, `secp.verify_schnorr(&sig, msg, &pubkey)` failed the unit tests.

But, [below on line 517](https://github.com/bitcoindevkit/bdk/pull/718/files#diff-78369b4827c629390df1a56a9ec4cae57346aa2f03cb66bcf9455cf75411923cL517), we attach the resulting signature to `pubkey`.

That seems wrong. Is that a bug?

Reply:
When doing a key-spend the key is tweaked with the merkle root (line 501-503), so the public key of the `keypair` variable is gonna be different from the "original" pubkey.

That said, key-spend signatures are stored in the `tap_key_sig` psbt field, which is not a map and thus doesn't contain the pubkey at all (unlike `tap_script_sigs`)